From e21c9c07ab9f25ec203c5a47a12316a46ca835b7 Mon Sep 17 00:00:00 2001
From: Bryan Cantrill <bryan@joyent.com>
Date: Wed, 6 Feb 2019 07:19:44 +0000
Subject: [PATCH] OS-7535 allow platform builds on base-64 image Reviewed by:
 Robert Mustacchi <rm@joyent.com> Reviewed by: Jonathan Perkin
 <jonathan.perkin@joyent.com> Approved by: Robert Mustacchi <rm@joyent.com>

---
 Makefile.gcc      | 6 ++++++
 binutils/Makefile | 2 ++
 node.js/Makefile  | 5 +++--
 3 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/Makefile.gcc b/Makefile.gcc
index 754dc83..b1e698a 100644
--- a/Makefile.gcc
+++ b/Makefile.gcc
@@ -27,7 +27,13 @@ PREFIX = /usr/gcc/$(GCC_VER)
 
 AUTOCONF_PREFIX = $(PREFIX)
 
+#
+# We set --enable-bootstrap to allow the build to mismatch the host
+#
 AUTOCONF_OPTS += \
+	--enable-bootstrap \
+	--build=i386-pc-solaris2.11 \
+	--host=i386-pc-solaris2.11 \
 	--with-ld=/usr/bin/ld \
 	--without-gnu-ld \
 	--with-gnu-as \
diff --git a/binutils/Makefile b/binutils/Makefile
index d956806..cbfe686 100644
--- a/binutils/Makefile
+++ b/binutils/Makefile
@@ -34,6 +34,8 @@ CFLAGS += $(CPPFLAGS)
 TARBALL =		$(VER).tar.bz2
 TARBALL_COMPRESS =	-j
 AUTOCONF_OPTS = \
+	--build=i386-pc-solaris2.11 \
+	--host=i386-pc-solaris2.11 \
 	--prefix=/usr/gnu \
 	--enable-64-bit-bfd \
 	--program-prefix=g
diff --git a/node.js/Makefile b/node.js/Makefile
index 0724322..2edc13e 100644
--- a/node.js/Makefile
+++ b/node.js/Makefile
@@ -18,7 +18,7 @@
 #
 # CDDL HEADER END
 #
-# Copyright 2018 Joyent, Inc.
+# Copyright 2019 Joyent, Inc.
 #
 
 MAJOR_VER =	0.10
@@ -33,7 +33,7 @@ VERSIONJS =	$(NODE_ROOT)/node_modules/platform_node_version.js
 # ../deps/v8/src/objects.h:5188:44: error: left operand of shift expression
 # '(-1 << 3)' is negative
 #
-CXXFLAGS +=	-fpermissive
+CXXFLAGS += -fpermissive
 
 #
 # https://github.com/nodejs/node/issues/6272
@@ -81,6 +81,7 @@ AUTOCONF_OPTS += \
 AUTOCONF_CFLAGS =	CFLAGS="$(CPPFLAGS) $(CFLAGS)"
 AUTOCONF_LIBS =
 AUTOCONF_ENV +=		CXXFLAGS="$(CPPFLAGS) $(CFLAGS) $(CXXFLAGS)"
+AUTOCONF_ENV +=		CXXFLAGS.host="$(CPPFLAGS) $(CFLAGS) $(CXXFLAGS)"
 
 #
 # Jump through hoops to get the locally-run build tools to build with the
-- 
2.21.0

